src/platform/DeviceInfoProvider.cpp

```diff
@@ -82,11 +82,16 @@ CHIP_ERROR DeviceInfoProvider::AppendUserLabel(EndpointId endpoint, const UserLa
 {
     size_t length;
 
-    // Increase the size of UserLabelList by 1
+    // Fetch current list length
     ReturnErrorOnFailure(GetUserLabelLength(endpoint, length));
-    ReturnErrorOnFailure(SetUserLabelLength(endpoint, length + 1));
 
-    // Append the user label at the end of UserLabelList
+    if (length >= kMaxUserLabelListLength)
+    {
+        return CHIP_ERROR_NO_MEMORY;
```

Comment:
Will this correctly get mapped to RESOURCE_EXHAUSTED or some similar error code by the callers?  The exact error codes for this API should be documented, and callers should be checked to make sure they handle them properly.

I'm OK with that happening in a followup, but please make sure it happens.